KitePrintSDK/src/main/java/ly/kite/print/ProductLoader.java

```diff
@@ -0,0 +1,1004 @@
+ * to place orders with any competitors to Kite Tech Ltd, all orders MUST go through the
+ * Kite Tech Ltd platform servers.
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ *****************************************************/
+
+///// Package Declaration /////
+
+package ly.kite.print;
+
+
+///// Import(s) /////
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.SystemClock;
+import android.util.Log;
+import android.util.Pair;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.math.BigDecimal;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Currency;
+import java.util.HashMap;
+import java.util.Iterator;
+
+import ly.kite.KiteSDKException;
+import ly.kite.KiteSDK;
+import ly.kite.product.MultipleCurrencyCost;
+import ly.kite.product.MultipleDestinationShippingCosts;
+import ly.kite.product.SingleCurrencyCost;
+import ly.kite.product.UserJourneyCoordinator;
+import ly.kite.product.UserJourneyType;
+
+
+///// Class Declaration /////
+
+/****************************************************
+ *
+ * This singleton class retrieves and stores all the
+ * product groups and products from the server.
+ *
+ ****************************************************/
+public class ProductLoader implements BaseRequest.BaseRequestListener
+  {
+  ////////// Static Constant(s) //////////
+
+  @SuppressWarnings("unused")
+  private static final String  LOG_TAG                               = "ProductLoader";
+
+  private static final boolean DISPLAY_DEBUGGING                     = false;
+
+  public  static final long    ANY_AGE_OK                            = -1;
+
+  private static final String  REQUEST_FORMAT_STRING                 = "%s/template/?limit=100";
+
+  private static final String  JSON_NAME_AMOUNT                      = "amount";
+  private static final String  JSON_NAME_CURRENCY                    = "currency";
+  private static final String  JSON_NAME_CENTIMETERS                 = "cm";
+  private static final String  JSON_NAME_COST                        = "cost";
+  private static final String  JSON_NAME_FORMATTED_AMOUNT            = "formatted";
+  private static final String  JSON_NAME_GROUP_IMAGE                 = "ios_sdk_class_photo";
```

Comment:
yuck there's definitely more than a few fields that are named badly in the response with iOS specific bits. mid term I want to introduce a nice public RESTful API interface that we're not ashamed of that will address a bunch of things